cpp/src/structs/utilities.cpp

```diff
@@ -341,6 +341,14 @@ void superimpose_parent_nulls(bitmask_type const* parent_null_mask,
   }
 }
 
+bool contains_list(column_view const& col)
+{
+  return col.type().id() == type_id::LIST ||
```

Comment:
Is this the right logic here?  If something _is_ a list column, it doesn't necessarily mean it contains a list column.  I think there are cases where we want to make sure "I can do X on this type, but not if it contains a list deeper down" where the top level type is actually a list :  for example sorting.   If this handled both list-containing-another-list as well as struct-containing-a-list, we could maybe make this function available as a more general utility.

Reply:
It's now redundant, found another function `is_or_has_nested_lists` in anonymous namespace that I moved to detail.